### PR TITLE
feat: Integrate RTS migration API in get page API

### LINKED ISSUE #26980: Integrate On demand migration API with Java server
```[tasklist]
### Tasks
- [ ] Integrate RTS API to apply migration
- [ ] Verify the migration with different test cases
- [ ] Add Junit tests for DSL Migration validation on server
- [ ] Update page apis on Java Server to serve latest DSL always
```

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
This PR calls the migration API on RTS to migrate the page DSL. 

#### PR fixes following issue(s)
Fixes #26980

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/DSLMigrationUtils.java [hunk 1]
```diff
@@ -30,32 +35,43 @@ public class DSLMigrationUtils {
...
+        ParameterizedTypeReference<ResponseDTO<DslVersionDTO>> parameterizedTypeReference =
+                new ParameterizedTypeReference<>() {};
+        return webClient
+                .get()
+                .uri(commonConfig.getRtsBaseUrl() + "/rts-api/v1/dsl/version")
+                .retrieve()
+                .bodyToMono(parameterizedTypeReference)
+                .map(responseDTO -> {
+                    int latestDslVersion = responseDTO.getData().getVersion();
+                    return pageDslVersion < latestDslVersion;
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/DSLMigrationUtils.java:48]: Can we add a error handler here? If the API fails we should send the DSL right

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java [hunk 2]
```diff
@@ -273,23 +276,75 @@ public Mono<PageDTO> getPage(String pageId, boolean viewMode) {
...
                 .map(responseUtils::updatePageDTOWithDefaultResources);
     }
 
+    private Mono<PageDTO> migrateAndUpdatePageDsl(NewPage newPage, PageDTO page, boolean viewMode) {
+        return dslMigrationUtils
+                .getLatestDslVersion()
+                .onErrorMap(throwable -> {
+                    log.error("Error fetching latest DSL version", throwable);
+                    return new AppsmithException(AppsmithError.RTS_SERVER_ERROR, "Error fetching latest DSL version");
+                })
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java:311]: Can we make the error message a more meaningful? Other would be throwing the 500 ISE error. This does not really help the user.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java [hunk 3]
```diff
@@ -273,23 +276,75 @@ public Mono<PageDTO> getPage(String pageId, boolean viewMode) {
...
+                .flatMap(latestDslVersion -> {
+                    // ensuring that the page has only one layout, as we don't support multiple layouts yet
+                    // when multiple layouts are supported, this code will have to be updated
+                    assert page.getLayouts().size() == 1;
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java:315]: QQ - Why are we asserting this?

**Review by AnaghHegde — COMMENTED** [review]
Minor nit, rest looks good to me.

**Comment by nayan-rafiq** [comment]
Closing this as it's already merged with another PR https://github.com/appsmithorg/appsmith/pull/28518

**Pull request closed without merging**